File: C/beginner/1164.c
// https://judge.beecrowd.com/pt/problems/view/1164
// Número Perfeito

// Libraries
#include <stdio.h>

// Main
int main(){

    // Variables
    int testCases, usrInput;

    // Getting number of cases
    scanf("%d", &testCases);

    // Iterating
    for(int i = 1; i <= testCases; i++){
        
        // Collecting candidates to check their perfection
        scanf("%d", &usrInput);
        
        // Starting sum, for checking perfection
        int sum = 0;

        // Iterating between multiples
        for(int j = usrInput - 1; j > 0; j--){
            if(usrInput % j == 0){
                sum += j;
            }
        }

        // Checking perfection
        if(sum == usrInput){
            printf("%d eh perfeito\n", usrInput);
        } else {
            printf("%d nao eh perfeito\n", usrInput);
        }
    }

    return 0;
}

// By: Fernando Serra
// https://github.com/fernandosserra

File: Python/beginner/1164.py
# https://judge.beecrowd.com/pt/problems/view/1164
# Número perfeito

# Getting test cases
test_cases = int(input())

# Function to find the perfection of a number
def is_perfect(n):
    divisors = [x for x in range(1, n // 2 + 1) if n % x == 0]
    return sum(divisors) == n

# Iterating between cases
for _ in range(test_cases):
    # Getting number
    number = int(input())
    
    # Showing result
    if is_perfect(number):
        print(f'{number} eh perfeito')
    else:
        print(f'{number} nao eh perfeito')
        
# By Fernando Serra
# https://github.com/fernandosserra